meilisearch/task.py

```diff
@@ -11,158 +11,147 @@
-    interval_in_ms (optional):
-        Time interval the method should wait (sleep) between requests.
-
-    Returns
-    -------
-    task:
-        Dictionary containing information about the processed asynchronous task.
-
-    Raises
-    ------
-    MeiliSearchTimeoutError
-        An error containing details about why Meilisearch can't process your request. Meilisearch error codes are described here: https://docs.meilisearch.com/errors/#meilisearch-errors
-    """
-    start_time = datetime.now()
-    elapsed_time = 0.0
-    while elapsed_time < timeout_in_ms:
-        task = get_task(config, uid)
-        if task["status"] not in ("enqueued", "processing"):
-            return task
-        sleep(interval_in_ms / 1000)
-        time_delta = datetime.now() - start_time
-        elapsed_time = time_delta.seconds * 1000 + time_delta.microseconds / 1000
-    raise MeiliSearchTimeoutError(
-        f"timeout of ${timeout_in_ms}ms has exceeded on process ${uid} when waiting for task to be resolve."
-    )
+class TaskHandler:
+    def __init__(self, config: Config):
+        self.config = config
+        self.http = HttpRequests(config)
+
+    def get_tasks(
+        self, parameters: Optional[Dict[str, Any]] = None
+    ) -> Dict[str, List[Dict[str, Any]]]:
+        """Get all tasks.
+
+        Parameters
+        ----------
+        parameters (optional):
+            parameters accepted by the get tasks route: https://docs.meilisearch.com/reference/api/tasks.html#get-tasks.
+
+        Returns
+        -------
+        task:
+            Dictionary with limit, from, next and results containing a list of all enqueued, processing, succeeded or failed tasks.
+
+        Raises
+        ------
+        MeiliSearchApiError
+            An error containing details about why Meilisearch can't process your request. Meilisearch error codes are described here: https://docs.meilisearch.com/errors/#meilisearch-errors
+        """
+        if parameters is None:
+            parameters = {}
+        for param in parameters:
+            if isinstance(parameters[param], list):
+                parameters[param] = ",".join(parameters[param])
+        return self.http.get(f"{self.config.paths.task}?{parse.urlencode(parameters)}")
+
+    def get_task(self, uid: int) -> Dict[str, Any]:
+        """Get one task.
+
+        Parameters
+        ----------
+        uid:
+            Identifier of the task.
+
+        Returns
+        -------
+        task:
+            Dictionary containing information about the status of the asynchronous task.
+
+        Raises
+        ------
+        MeiliSearchApiError
+            An error containing details about why Meilisearch can't process your request. Meilisearch error codes are described here: https://docs.meilisearch.com/errors/#meilisearch-errors
+        """
+        return self.http.get(f"{self.config.paths.task}/{uid}")
+
+    def cancel_tasks(self, parameters: Dict[str, Any]) -> TaskInfo:
```

Comment:
I know you didn't make a change here, but I noticed the type hint for `parameters` here is incorrect. It should be:

```py
    def cancel_tasks(self, parameters: Union[Dict[str, Any], None] = None) -> TaskInfo:
```

Reply:
Noted

Reply:
I think `Optional` would be even better.

```python
    def cancel_tasks(self, parameters: Optional[Dict[str, Any]] = None) -> TaskInfo: 
```

Reply:
I thought I saw the other style being used, but you are right that this one is consistent with the others so better here.